app/filters/validate_schema.rb

```diff
@@ -0,0 +1,26 @@
+class SchemaNotFoundError < StandardError
+end
+
+class ValidateSchema
+  class << self
+    def before(controller)
+      schema = "request.#{controller.request.params['controller']}"
+      validate(controller.request.params, schema)
+    end
+
+    def validate(metadata, schema)
+      service_schema = find(schema)
+      JSON::Validator.validate!(service_schema, metadata)
+    end
+
+    def find(schema)
+      schema_file = schema.gsub('.', '/')
+      schema = File.read(File.join(
```

Comment:
Since the schemas won't change per service, is it a good idea to cache the File read into a constant or a class variable? *If yes, that's okay to do it in another PR I believe.

Reply:
so at this point the schema it is looking for is a request schema. there are currently two types, one for a service and one for a version. the request schemas reference the service schemas internally. we could cache the request.service schema and the request.version schema though . . .